client-spark/common/src/main/java/org/apache/spark/shuffle/RssSparkShuffleUtils.java

```diff
@@ -92,8 +92,9 @@ public static ShuffleManager loadShuffleManager(String name, SparkConf conf, boo
   public static List<CoordinatorClient> createCoordinatorClients(SparkConf sparkConf) {
     String clientType = sparkConf.get(RssSparkConfig.RSS_CLIENT_TYPE);
     String coordinators = sparkConf.get(RssSparkConfig.RSS_COORDINATOR_QUORUM);
-    CoordinatorClientFactory coordinatorClientFactory = new CoordinatorClientFactory(ClientType.valueOf(clientType));
-    return coordinatorClientFactory.createCoordinatorClient(coordinators);
+    CoordinatorClientFactory factory = CoordinatorClientFactory.getInstance();
```

Comment:
I prefer the following code


```java
 CoordinatorClientFactory factory = CoordinatorClientFactory.getInstance();
 this.coordinatorClients = factory.getOrCreateCoordinatorClient(conf.get(ShuffleServerConf.RSS_CLIENT_TYPE), this.coordinatorQuorum);
```

And I think we also need to hold the client into the cache to avoid recreating the connection. Right?

Reply:
Hi @zuston 
Good catch! Thanks for your kind reminder and I will handle it. :smile:

Reply:
Hi @zuston 

Sorry for the late reply! I updated the part you mentioned but encountered some tricky errors. It would be appreciated if you are available to help me. :)